packages/experiments/src/components/Button/Button.types.tsx

```diff
@@ -9,16 +16,28 @@ export type IButtonStates = 'baseState' | 'enabled' | 'disabled' | 'expanded';
 
 export type IButtonVariants = 'baseVariant' | 'primary' | 'circular';
 
-export type IButtonSlots = 'root' | 'stack' | 'text' | 'icon' | 'menuIcon';
+export interface IButtonSlots {
```

Comment:
I like how these slots have their own interface. I think that's an important distinction to make, especially when it comes to documenting the control. This should definitely be the way that devs write controls.

Interesting though that this means there "will" be a 'root' prop. Interesting to note that 'root' is not the IButtonProps interface, but the DOM element that is used for the outer parent. We should certainly consider situations where 'root' might get hard to determine. Like panel. Is root the actual panel, or the container around the layer? What happens if the root DOM element changes based on some prop (from <button> to <div> wrapping a button), or maybe we try to avoid that.

Reply:
now i think about it... could there ever be a component that is actually rooted as a fragment?

Reply:
yeah, that's certainly a possibility.

Reply:
Do Fragments have DOM representation? Can they have a className applied to them? The whole point of React Fragments is not to have any DOM representation, so I don't think either of these is true and I don't think they would work as root elements for us from a styling perspective.

Reply:
This is pretty awesome